hail/src/main/scala/is/hail/expr/ir/TableValue.scala

```diff
@@ -157,4 +157,37 @@ case class TableValue(typ: TableType, globals: BroadcastRow, rvd: RVD) {
       rvd.toRows,
       typ.rowType.schema.asInstanceOf[StructType])
   }
+
+  def toMatrixValue(colsFieldName: String, entriesFieldName: String, colKey: IndexedSeq[String]): MatrixValue = {
+
+    val (colType, colsFieldIdx) = typ.globalType.field(colsFieldName) match {
+      case Field(_, TArray(t@TStruct(_, _), _), idx) => (t, idx)
+      case Field(_, t, _) => fatal(s"expected cols field to be an array of structs, found $t")
+    }
+    val m = Map(entriesFieldName -> MatrixType.entriesIdentifier)
+
+    val newRowType = typ.rowType.rename(m)
+
+    val mType: MatrixType = MatrixType(
+      typ.globalType.deleteKey(colsFieldName, colsFieldIdx),
+      colKey,
+      colType,
+      typ.key,
+      newRowType)
+
+    val colValues = globals.value.getAs[IndexedSeq[Annotation]](colsFieldIdx)
+    val newGlobals = {
+      val (pre, post) = globals.value.toSeq.splitAt(colsFieldIdx)
+      Row.fromSeq(pre ++ post.tail)
+    }
+
+    val newRVD = rvd.cast(rvd.rowPType.rename(m))
```

Comment:
Where will `toMatrixValue` be used? Should it map over the rvd to add checks that entry arrays are non-missing and the same length as `colValues`?

Reply:
Used in #5785 #5787 

This should not check, because it will be called from internal lowering stages, not user code. CastTableToMatrix still does check.

Reply:
(though I think that check should be implemented in Python, not scala)